src/test/enhetstester/kotlin/no/nav/familie/ks/sak/sikkerhet/TilgangServiceTest.kt

```diff
@@ -0,0 +1,502 @@
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.slf4j.MDC
+import org.springframework.cache.CacheManager
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager
+import java.time.LocalDate
+
+@ExtendWith(MockKExtension::class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class TilgangServiceTest {
+
+    @MockK
+    private lateinit var integrasjonService: IntegrasjonService
+
+    @MockK
+    private lateinit var behandlingRepository: BehandlingRepository
+
+    @MockK
+    private lateinit var fagsakService: FagsakService
+
+    @MockK
+    private lateinit var personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository
+
+    private val cacheManager = ConcurrentMapCacheManager()
+
+    private val rolleConfig = RolleConfig(
+        BehandlerRolle.BESLUTTER.name,
+        BehandlerRolle.SAKSBEHANDLER.name,
+        BehandlerRolle.VEILEDER.name,
+        KODE6 = "kode6",
+        KODE7 = "kode7"
+    )
+    private val auditLogger = AuditLogger("familie-ks-sak")
+    private lateinit var tilgangService: TilgangService
+
+    private val fagsak = defaultFagsak()
+    private val behandling = lagBehandling(fagsak)
+    private val aktør = fagsak.aktør
+    private val personopplysningGrunnlag = lagPersonopplysningGrunnlag(
+        behandlingId = behandling.id,
+        søkerPersonIdent = aktør.aktivFødselsnummer(),
+        barnasIdenter = emptyList()
+    )
+
+    @BeforeAll
+    fun beforeAll() {
+        tilgangService = TilgangService(
+            integrasjonService = integrasjonService,
+            behandlingRepository = behandlingRepository,
+            personopplysningGrunnlagRepository = personopplysningGrunnlagRepository,
+            fagsakService = fagsakService,
+            rolleConfig = rolleConfig,
+            cacheManager = cacheManager,
+            auditLogger = auditLogger
+        )
+    }
+
+    @BeforeEach
+    internal fun beforeEach() {
+        clearAllMocks()
+        clearCache(cacheManager)
+        MDC.put(MDCConstants.MDC_CALL_ID, "00001111")
+        mockBrukerContext(groups = listOf(BehandlerRolle.SAKSBEHANDLER.name))
+        every { fagsakService.hentFagsak(fagsak.id) } returns fagsak
+        every { behandlingRepository.finnAktivBehandling(any()) } returns behandling
+        every { behandlingRepository.finnBehandlinger(fagsak.id) } returns listOf(behandling)
+        every { personopplysningGrunnlagRepository.findByBehandlingAndAktiv(any()) } returns personopplysningGrunnlag
+    }
+
+    @AfterEach
+    internal fun tearDown() {
+        clearBrukerContext()
+    }
+
+    private fun clearCache(cacheManager: CacheManager) {
+        cacheManager.cacheNames.parallelStream().forEach { name: String? ->
+            cacheManager.getCache(
+                name!!
+            )!!.clear()
+        }
+    }
```

Comment:
```suggestion
    private fun clearCache(cacheManager: CacheManager) =
        cacheManager.cacheNames.parallelStream().forEach {
            cacheManager.getCache(it)?.clear()
        }
```
Kan vi gjøre noe sånt istedenfor, gitt at navnet ikke kan være null? 😊